doc/05_Objects/01_Object_Classes/05_Class_Settings/50_Path_Formatter.md

```diff
@@ -45,31 +45,30 @@ class TheFormatter implements PathFormatterInterface
      */
     public function formatPath(array $result, ElementInterface $source, array $targets, array $params): array
     {
-        /** @var  $fd Data */
-        $fd = $params["fd"];
-        $context = $params["context"];
+        /** @var Data $fd */
+        $fd = $params['fd'];
+        $context = $params['context'];
 
         foreach ($targets as $key => $item) {
-            $newPath = $item["fullpath"] .  " - " . time();
-            if ($context["language"]) {
-                $newPath .= " " . $context["language"];
+            $newPath = ($item['path'] ?? $item['fullpath']) . ' - ' . time();
+            if (isset($context['language'])) {
+                $newPath .= ' ' . $context['language'];
             }
 
-            if ($item["type"] == "object") {
-                $targetObject = Concrete::getById($item["id"]);
+            if ($item['type'] === 'object') {
+                $targetObject = Concrete::getById($item['id']);
                 if ($targetObject instanceof News) {
-                    $newPath = $targetObject->getTitle() . " - " . $targetObject->getShortText();
+                    $newPath = $targetObject->getTitle() . ' - ' . $targetObject->getShortText();
                 }  else if ($targetObject instanceof BlogArticle) {
                     $newPath = $targetObject->getTitle();
                 }
-            } elseif ($item["type"] == "asset") {
-                $asset = Asset::getById($item["id"]);
+            } elseif ($item['type'] === 'asset') {
+                $asset = Asset::getById($item['id']);
                 if ($asset) {
-                    $title = $asset->getMetadata("title");
-
+                    $title = $asset->getMetadata('title');
 
                     if (!$title) {
-                        $title = "this guy does not have a title, use " . $newPath . " instead";
+                        $title = 'this guy does not have a title, use ' . $newPath . ' instead';
```

Comment:
Thank you for your PR!

Just casually noticed this funny legacy line that could be changed now while we are it 😄 the rest LGTM 
```suggestion
                        $title = 'this asset does not have a title, use ' . $newPath . ' instead';
```